crates/store/re_data_loader/src/mcap/schema/sensor_msgs/joint_state.rs

```diff
@@ -0,0 +1,144 @@
+use re_chunk::{
+    Chunk, ChunkId,
+    external::arrow::array::{Float64Builder, ListBuilder, StringBuilder},
+};
+use re_log_types::TimeCell;
+use re_mcap_ros2::sensor_msgs;
+use re_sorbet::SorbetSchema;
+use re_types::archetypes::{Scalars, SeriesLines};
+
+use crate::mcap::{
+    cdr,
+    decode::{McapMessageParser, ParserContext, PluginError, SchemaName, SchemaPlugin},
+};
+
+/// Plugin that parses `sensor_msgs/msg/JointState` messages.
+#[derive(Default)]
+pub struct JointStateSchemaPlugin;
+
+impl SchemaPlugin for JointStateSchemaPlugin {
+    fn name(&self) -> SchemaName {
+        "sensor_msgs/msg/JointState".into()
+    }
+
+    fn parse_schema(&self, _channel: &mcap::Channel<'_>) -> Result<SorbetSchema, PluginError> {
+        Err(PluginError::Other(anyhow::anyhow!("todo")))
+    }
+
+    fn create_message_parser(
+        &self,
+        _channel: &mcap::Channel<'_>,
+        num_rows: usize,
+    ) -> Box<dyn McapMessageParser> {
+        Box::new(JointStateMessageParser::new(num_rows)) as Box<dyn McapMessageParser>
+    }
+}
+
+pub struct JointStateMessageParser {
+    joint_names: ListBuilder<StringBuilder>,
+    positions: ListBuilder<Float64Builder>,
+    velocities: ListBuilder<Float64Builder>,
+    efforts: ListBuilder<Float64Builder>,
+}
+
+impl JointStateMessageParser {
+    /// Create a new [`JointStateMessageParser`]
+    pub fn new(num_rows: usize) -> Self {
+        Self {
+            joint_names: ListBuilder::with_capacity(StringBuilder::new(), num_rows),
+            positions: ListBuilder::with_capacity(Float64Builder::new(), num_rows),
+            velocities: ListBuilder::with_capacity(Float64Builder::new(), num_rows),
+            efforts: ListBuilder::with_capacity(Float64Builder::new(), num_rows),
+        }
+    }
+}
+
+impl McapMessageParser for JointStateMessageParser {
+    fn append(&mut self, ctx: &mut ParserContext, msg: &mcap::Message<'_>) -> anyhow::Result<()> {
+        let joint_state = cdr::try_decode_message::<sensor_msgs::JointState>(msg.data.as_ref())
+            .map_err(|err| PluginError::Other(anyhow::anyhow!(err)))?;
+
```

Comment:
use the destructor pattern to make sure we don't forget any field in `JointState`